packages/core/src/shared/plugins/suggestion/SuggestionPlugin.ts

```diff
@@ -85,6 +95,111 @@ export function findCommandBeforeCursor(
   };
 }
 
+class SuggestionPluginView<T extends SuggestionItem> {
+  editor: Editor;
+  pluginKey: PluginKey;
+
+  itemCallback: (props: { item: T; editor: Editor; range: Range }) => void;
+
+  suggestionsMenuParams: SuggestionsMenuParams<T>;
+  suggestionsMenu: SuggestionsMenu<T>;
+
+  constructor({
+    editor,
+    pluginKey,
+    onSelectItem: selectItemCallback = () => {},
+    suggestionsMenuFactory,
+  }: SuggestionPluginViewOptions<T>) {
+    this.editor = editor;
+    this.pluginKey = pluginKey;
+
+    this.itemCallback = selectItemCallback;
+
+    this.suggestionsMenuParams = this.initSuggestionsMenuParams();
+    this.suggestionsMenu = suggestionsMenuFactory(this.suggestionsMenuParams);
+  }
+
+  update(view: EditorView, prevState: EditorState) {
+    const prev = this.pluginKey.getState(prevState);
+    const next = this.pluginKey.getState(view.state);
+
+    // See how the state changed
+    const started = !prev.active && next.active;
+    const stopped = prev.active && !next.active;
+    // TODO: Currently also true for cases in which an update isn't needed so selected list item index updates still
+    //  cause the view to update. May need to be more strict.
+    const changed = prev.active && next.active;
+
+    // Cancel when suggestion isn't active
+    if (!started && !changed && !stopped) {
+      return;
+    }
+
+    const state = stopped ? prev : next;
+
+    if (stopped) {
+      this.suggestionsMenu.hide();
+
+      // Listener stops focus moving to the menu on click.
+      this.suggestionsMenu.element!.removeEventListener("mousedown", (event) =>
+        event.preventDefault()
+      );
+    }
+
+    if (changed) {
+      this.updateSuggestionsMenuParams(state);
+      this.suggestionsMenu.update(this.suggestionsMenuParams);
+    }
+
+    if (started) {
+      this.updateSuggestionsMenuParams(state);
+      this.suggestionsMenu.show(this.suggestionsMenuParams);
+
+      // Listener stops focus moving to the menu on click.
+      this.suggestionsMenu.element!.addEventListener("mousedown", (event) =>
+        event.preventDefault()
+      );
+    }
+  }
+
+  initSuggestionsMenuParams() {
```

Comment:
Ideally, I think we should also get rid of this. This relates to a bigger issue with the Factories; we now need code to initialize "fake data", which is not used anyway. We discussed this before, but while review this (plugin -> class conversion) it now becomes clear it's annoying. More a question of _when_ than _if_ we want to fix it imo